tests/unit/CMakeLists.txt

```diff
@@ -13,8 +13,8 @@ include(CTest)
 
 set(CMAKE_CXX_STANDARD 20)
 
-set(CMAKE_CXX_FLAGS_DEBUG "-g")
-set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+set(CMAKE_CXX_FLAGS_DEBUG "-g -pthread")
+set(CMAKE_CXX_FLAGS_RELEASE "-O3 -pthread")
```

Comment:
are we using pthreads here?

Reply:
I thought it was necessary for using the `Thread` object in the test but looks like it doesn't. removing...